nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -90,4 +119,162 @@ private static void invalidInstantiationError(
         errorBuilder.createErrorDescription(
             errorMessage, analysis.buildDescription(tree), state, null));
   }
+
+  @SuppressWarnings("UnusedVariable")
+  public static void checkInstantiationForAssignments(
+      AssignmentTree tree, Config config, VisitorState state, NullAway analysis) {
```

Comment:
We will need to generalize this check to more than `AssignmentTree`s; e.g., we need to handle initializers in `VariableTree`s.  To do so we may need to generalize this method's parameters, or create another method for `VariableTree`s.  For a `VariableTree` `t` (e.g., `Foo<@Nullable String> f = new Foo<@Nullable String>();`), we should do a check if `t.getInitializer()` is not `null`.  Then, the LHS type comes from `t.getType()`.  For the cases we care about this will likely be a `ParameterizedTypeTree`; we'll have to get if the generic type annotations are properly present.